lib/protobuf/field/message_field.rb

```diff
@@ -9,11 +9,7 @@ class MessageField < BaseField
       #
 
       def acceptable?(val)
-        unless val.is_a?(type_class) || val.respond_to?(:to_hash)
-          fail TypeError, "Expected value of type '#{type_class}' for field #{name}, but got '#{val.class}'"
```

Comment:
I think this is a good change, but wanted to call this out as a behavior change since you mentioned you were trying to maintain existing behavior.